sklearn/svm/_base.py

```diff
@@ -489,10 +490,6 @@ def _validate_for_predict(self, X):
                 raise ValueError("X.shape[1] = %d should be equal to %d, "
                                  "the number of samples at training time" %
                                  (X.shape[1], self.shape_fit_[0]))
-        elif not callable(self.kernel) and X.shape[1] != self.shape_fit_[1]:
```

Comment:
do we still need `shape_fit_` then?

Reply:
`shape_fit_[0]` is still being used in some places of the codebase.